src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -60,10 +64,28 @@ public JobService(IamService samService,
         this.appConfig = appConfig;
         this.stairwayJdbcConfiguration = stairwayJdbcConfiguration;
         this.migrateConfiguration = migrateConfiguration;
+        this.isShutdown = new AtomicBoolean(false);
 
         ExecutorService executorService = Executors.newFixedThreadPool(appConfig.getMaxStairwayThreads());
         ExceptionSerializer serializer = new StairwayExceptionSerializer(objectMapper);
-        stairway = new Stairway(executorService, applicationContext, serializer);
+        stairway = new Stairway(executorService, applicationContext, serializer, "stairwayname");
+    }
+
+    /**
+     * Stop accepting jobs and shutdown stairway
+     */
+    public void shutdown() throws InterruptedException {
+        isShutdown.set(true);
+
+        // We allocate 3/4 of the time for graceful shutdown. Then call terminate for the rest of the time.
+        int shutdownTimeout = appConfig.getShutdownTimeoutSeconds();
+        int gracefulTimeout = (shutdownTimeout * 3) / 4;
+        int terminateTimeout = (shutdownTimeout - gracefulTimeout) - 2;
```

Comment:
What happens if `terminateTimeout` < 0?

Reply:
Good point. I added a floor, so we will never use too small a value.